plugins/catalog-backend/src/ingestion/processors/AzureApiReaderProcessor.ts

```diff
@@ -18,9 +18,17 @@ import { LocationSpec } from '@backstage/catalog-model';
 import fetch, { RequestInit, HeadersInit } from 'node-fetch';
 import * as result from './results';
 import { LocationProcessor, LocationProcessorEmit } from './types';
+import { Config } from '@backstage/config';
 
 export class AzureApiReaderProcessor implements LocationProcessor {
-  private privateToken: string = process.env.AZURE_PRIVATE_TOKEN || '';
+  private privateToken: string;
+
+  constructor(config: Config) {
+    this.privateToken =
+      (config.getOptional(
```

Comment:
There's a getOptionalString right

Reply:
Same for all other places below

Reply:
`getOptionalString` throws exception in case of empty strings `Invalid type in config for key 'catalog.processors.azureApi.privateToken' in '', got empty-string, wanted string` . and we had tests for all processors when the private token is empty string the token is not sent or empty string is sent . did not change the behavior. if you think it should not be empty and its fine to change the current behavior i will remove those tests and use `getOptionalString`

Reply:
Oh I see. The end result I was after is that the type should be verified (that giving an array is a failure for example). If it also rejects empty strings then in a sense, that could be considered a feature because it IS an illegal value in this case, but I presume that the tests to accept it are there as a convenience - that we can have a default config that fetches env vars, and if those vars aren't given, we just act as if nothing happened instead of exploding.

@Rugvip how do you feel about the empty-is-failure behavior, could we add an allowEmpty param or something at some later point? :)

Reply:
@freben yes empty string is an illegal value in this case. it should be either undefined or a non empty string.
I have updated processors to use `getOptionalString` and corresponding tests. if env vars are not given, everything is fine because they are undefined instead of empty string.